app/src/main/java/com/duckduckgo/app/browser/refreshpixels/RefreshPixelSender.kt

```diff
@@ -19,82 +19,72 @@ package com.duckduckgo.app.browser.refreshpixels
 import com.duckduckgo.app.browser.customtabs.CustomTabPixelNames
 import com.duckduckgo.app.di.AppCoroutineScope
 import com.duckduckgo.app.pixels.AppPixelName
-import com.duckduckgo.app.pixels.AppPixelName.RELOAD_THREE_TIMES_WITHIN_20_SECONDS
-import com.duckduckgo.app.pixels.AppPixelName.RELOAD_TWICE_WITHIN_12_SECONDS
 import com.duckduckgo.app.statistics.pixels.Pixel
 import com.duckduckgo.app.statistics.pixels.Pixel.PixelType.Daily
 import com.duckduckgo.app.trackerdetection.blocklist.BlockListPixelsPlugin
 import com.duckduckgo.app.trackerdetection.blocklist.get2XRefresh
 import com.duckduckgo.app.trackerdetection.blocklist.get3XRefresh
-import com.duckduckgo.common.utils.CurrentTimeProvider
 import com.duckduckgo.common.utils.DispatcherProvider
 import com.duckduckgo.di.scopes.AppScope
 import com.squareup.anvil.annotations.ContributesBinding
 import dagger.SingleInstanceIn
 import javax.inject.Inject
 import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.launch
+import timber.log.Timber
 
 interface RefreshPixelSender {
     fun sendMenuRefreshPixels()
     fun sendCustomTabRefreshPixel()
     fun sendPullToRefreshPixels()
+    fun sendBreakageRefreshPixels(patternsDetected: Set<Int>)
 }
 
 @ContributesBinding(AppScope::class)
 @SingleInstanceIn(AppScope::class)
 class DuckDuckGoRefreshPixelSender @Inject constructor(
     private val pixel: Pixel,
-    private val dao: RefreshDao,
-    private val currentTimeProvider: CurrentTimeProvider,
     private val blockListPixelsPlugin: BlockListPixelsPlugin,
     @AppCoroutineScope private val appCoroutineScope: CoroutineScope,
     private val dispatcherProvider: DispatcherProvider,
 ) : RefreshPixelSender {
 
     override fun sendMenuRefreshPixels() {
-        sendTimeBasedPixels()
         pixel.fire(AppPixelName.MENU_ACTION_REFRESH_PRESSED)
         pixel.fire(AppPixelName.REFRESH_ACTION_DAILY_PIXEL, type = Daily())
     }
 
     override fun sendPullToRefreshPixels() {
-        sendTimeBasedPixels()
         pixel.fire(AppPixelName.BROWSER_PULL_TO_REFRESH)
         pixel.fire(AppPixelName.REFRESH_ACTION_DAILY_PIXEL, type = Daily())
     }
 
     override fun sendCustomTabRefreshPixel() {
-        sendTimeBasedPixels()
         pixel.fire(CustomTabPixelNames.CUSTOM_TABS_MENU_REFRESH)
     }
 
-    private fun sendTimeBasedPixels() {
+    override fun sendBreakageRefreshPixels(patternsDetected: Set<Int>) {
```

Comment:
I would call this method something like `onRefreshPatternDetected`, the fact that internally we send a pixel or do something else is irrelevant to the caller so I wouldn't be that specific.